src/State.h

```diff
@@ -1,16 +1,20 @@
 #ifndef __STATE_H__
 #define __STATE_H__
 
-#include "stdlibs.h"
 #include "sexptypes.h"
+#include "stdlibs.h"
 
 using namespace std;
 
 #define RID_INVALID (rid_t) - 1
 
-const int RDT_SQL_LOOKUP_PROMISE = 0x0;
-const int RDT_SQL_LOOKUP_PROMISE_EXPRESSION = 0x1;
-const int RDT_SQL_FORCE_PROMISE = 0xF;
+enum class promise_event {
```

Comment:
Changing these values will break all my analyses - need to make sure the code for the analyses is updated.

Reply:
Same here, I have to update my analysis. I am not sure of a good way to resolve this. Unfortunately, there are events related to promises which have the same code. One option is to split them across different enums but then it would be arbitrary split. You suggest what you would much rather have.

Reply:
I don't know. I am just not looking forward to all my scripts breaking again. I guess leave it as it is.